python_training/tqdm.py

```diff
@@ -0,0 +1,88 @@
+from contextlib import contextmanager
+import sys
+import time
+
+
+class Tqdm():
+    """
+    Initiates a class that wraps iterable objects, and presents information regarding the iteration progress.
+    """
+    def __init__(self, iterable):
+        """
+        Initiates the wrapper class.
+        :param iterable: The iterable on which information will be provided during iteration.
+        """
+        self.iterable = iter(iterable)
+        self.max_iteration_number = len(iterable) - 1
```

Comment:
This is a little confusing (with the `-1` and then the `<=` later on).
Why not use `self.iterations_limit = len(iterable)` instead?